app/controllers/participations_controller.rb

```diff
@@ -10,9 +11,15 @@ def index
   def show
   end
 
-  # GET /participations/new
+  # GET /posts/:post_id/register
   def new
-    @participation = Participation.new
+    post = Post.find(params[:post_id])
+    if post.event.nil? || post.event.event_type.nil?
```

Comment:
This can be simplified with the safe navigation operator like:
```suggestion
    if post.event&.event_type.nil?
```